### PR TITLE
Override docker version deprecation

### PR DESCRIPTION
**Pull request by dperl-dls**
Temporary patch for #1199 to allow CI to run
Docs failure is because of https://github.com/bluesky/ophyd/issues/1197

### CODE REVIEW

**Review by mrakitin — COMMENTED** [review]
Thanks for the temporary fix, @dperl-dls!
The docs CI fails with:
```
intersphinx inventory 'https://blueskyproject.io/bluesky/objects.inv' not fetchable due to <class 'requests.exceptions.HTTPError'>: 404 Client Error: Not Found for url: https://blueskyproject.io/bluesky/objects.inv
```

I think we need to use https://blueskyproject.io/bluesky/main/objects.inv now. Do you mind changing the link [here](https://github.com/bluesky/ophyd/blob/1f8b64f52fcf99add0a50d67abdc69adc80b036f/docs/conf.py#L107) while you are here? Thanks!

File: setup.cfg [hunk 1]
```diff
@@ -20,7 +20,7 @@ setup_requires =
 # Specify any package dependencies below.
 install_requires =
     networkx>=2.0
-    numpy
+    numpy<2.0 # See https://github.com/mdavidsaver/p4p/issues/145
```
Comment (tacaswell) [inline setup.cfg:23]: Can we put this pinning in the pva section?

Now that we have split ophyd async out do we need p4p as a dep anymore?

Comment (dperl-dls) [reply setup.cfg]: Yeah that makes sense, I'll move it there. I don't know whether you need the pva section at all anymore but do feel like it's a bit of a separate issue to this.

**Review by mrakitin — APPROVED** [review]
Looks good to me, thanks for the efforts, @dperl-dls!

File: setup.cfg (hunk 1 above)
Comment (mrakitin) [reply setup.cfg]: Opened a separate issue for removing the PVA section: https://github.com/bluesky/ophyd/issues/1201.

**Pull request merged**